src/bootstrap/builder.rs

```diff
@@ -1079,7 +1079,7 @@ impl<'a> Builder<'a> {
         //
         // Only clear out the directory if we're compiling std; otherwise, we
         // should let Cargo take care of things for us (via depdep info)
```

Comment:
Please update this comment, too.

I am a little wary of making this change -- IIRC, and based on description of the commit that added this, it was quite intentional to limit it so harshly in #67760... but presuming we don't get a scattering of bug reports, it seems like we could do this.

I think one thing we can do, in theory, is not do this if we're not using incremental, but that seems worse to me right now. If we start getting bug reports due to extra rebuilds that users don't expect, then we can revisit this decision and figure out a more targeted fix. AFAICT, it should only be necessary to clear out the incremental state for rustc, while this targets all output directories (in build/check invocations). My guess is that will lead to some problems, but we'll see.

Reply:
> Please update this comment, too.

Sure, done.

> I think one thing we can do, in theory, is not do this if we're not using incremental, but that seems worse to me right now.

Hmm, that makes sense to me. Do you want me to do that now? I think the only case this can cause bugs is when using incremental; or I guess if cargo assumes that output artifacts are the same. Not sure when that could happen though.